### PR TITLE
Optimize android side gradle settings and clean source

### PR DESCRIPTION
**Pull request by xpwmaosldk**
Hi

It's my first step for this beautiful app.

Update gradle version and fix some gradle warring message.
Optimize android multidex settings(ref: https://developer.android.com/studio/build/multidex?#mdex-gradle)
Remove unused android files.
Remove unused dart import.

### CODE REVIEW

File: mobile/.metadata [hunk 1]
```diff
@@ -1,10 +1,45 @@
 # This file tracks properties of this Flutter project.
 # Used by Flutter tool to assess capabilities and perform upgrades etc.
 #
-# This file should be version controlled and should not be manually edited.
```
Comment (alextran1502) [inline mobile/.metadata:4]: Was this file changed manually? This line specifically said it should not be manually edited

Comment (xpwmaosldk) [reply mobile/.metadata:4]: No i didn't change manually

**Comment by alextran1502** [comment]
Hello, I've upgraded the build server to use JDK v11 and it seems like everything is passing. 

I have a question about styling the code, do you use a specific program to do that or you've done it manually?

File: mobile/pubspec.yaml [hunk 2]
```diff
@@ -43,10 +41,17 @@ dependencies:
   http: 0.13.4
   cancellation_token_http: ^1.1.0
 
+  path: ^1.8.1
+  path_provider: ^2.0.11
+  latlong2: ^0.8.1
+  collection: ^1.16.0
+  http_parser: ^4.0.1
```
Comment (alextran1502) [inline mobile/pubspec.yaml:48]: Hello, Why are these packages get added?

File: mobile/analysis_options.yaml [hunk 3]
```diff
@@ -24,6 +24,7 @@ linter:
   rules:
     # avoid_print: false  # Uncomment to disable the `avoid_print` rule
     # prefer_single_quotes: true  # Uncomment to enable the `prefer_single_quotes` rule
+    use_build_context_synchronously: false
```
Comment (alextran1502) [inline mobile/analysis_options.yaml:27]: Can you help me understand what does this line help us?

**Comment by alextran1502** [comment]
Thank you for the PR, I've taken a look and have a few comments, besides those. Everything is good to go.

**Comment by xpwmaosldk** [comment]
My code is automatically formatted by vscode.
The rule is probably a regular dart style guide

File: mobile/.metadata (hunk 1 above)
Comment (alextran1502) [reply mobile/.metadata:4]: Got it, thank you!

**Comment by alextran1502** [comment]
> It is an honor to be able to contribute to this project.

The honor is mine, my friend. Thank you for helping out 🥇

**Pull request merged**